src/components/Input/Input.tsx

```diff
@@ -67,11 +89,22 @@ class Input extends UIComponent<any, any> {
     type: 'text',
   }
 
+  static autoControlledProps = ['value']
+
   inputRef: any
 
-  computeTabIndex = props => {
-    if (!_.isNil(props.tabIndex)) return props.tabIndex
-    if (props.onClick) return 0
+  state: any = { value: this.props.value || '' }
+
+  getInitialAutoControlledState() {
+    return { value: '' }
+  }
+
+  handleChange = e => {
+    const value = _.get(e, 'target.value')
+
+    _.invoke(this.props, 'onChange', e, { ...this.props, value })
+
+    !this.props.value && this.setState({ value })
```

Comment:
`AutoControlledComponent` allows you to `this.trySetState()` which will automatically handle the proper checks against props of the same name as the state keys you are trying to set.